code/game/objects/items/weapons/tools.dm

```diff
@@ -728,14 +728,20 @@
 			var/transfer_amount = min(G.amount_per_transfer_from_this,space)
 			user.simple_message("<span class='info'>You transfer [transfer_amount] units to the [src].</span>",
 				"<span class='info'>The tool gulps down your drink!</span>")
-			G.reagents.trans_id_to(src,SACID,transfer_amount)
+			if(G.reagents.has_reagent(SACID, 1))
+				G.reagents.trans_id_to(src,SACID,transfer_amount)
+			else
+				G.reagents.trans_id_to(src,FORMIC_ACID,transfer_amount)
```

Comment:
Existing soldering iron code makes it not accept reagents from any beakers which have a mixture of chemicals, so it'd never get to this point if it has both.
I could change that condition if you want.